tiatoolbox/tools/patchextraction.py

```diff
@@ -109,6 +116,13 @@ def __init__(
         self.locations_df = None
         self.coordinate_list = None
         self.stride = None
+
+        assert (
```

Comment:
Please raise ValueError. Assert is not allowed in the code.

Reply:
On this, I believe you don't need to check for the type and range here as you are doing this in the `filter_coordinates_fast` function as well where I believe is the right place to do it. So, get rid of this check here and replace the one in `filter_coordinates_fast` with `ValueError`.

Furthermore, you need to add a test for this special case in the unit test, when an unacceptable value for `min_mask_ratio` is passed to the function. The test should be added somewhere around here:  
https://github.com/rogertrullo/tiatoolbox/blob/0ee90054fc4c4b88251be57b868f51d32554ab69/tests/test_patch_extraction.py#L456